engine-precompiles/src/random.rs

```diff
@@ -0,0 +1,85 @@
+use super::{EvmPrecompileResult, Precompile};
+#[cfg(feature = "contract")]
+use crate::prelude::sdk;
+use crate::prelude::Address;
+#[cfg(not(feature = "contract"))]
+use crate::prelude::H256;
+use crate::PrecompileOutput;
+use evm::{Context, ExitError};
+
+mod costs {
+    use crate::prelude::types::Gas;
+
+    // TODO(#51): Determine the correct amount of gas
+    pub(super) const RANDOM_BYTES_GAS: Gas = 0;
+}
+
+const ERR_TARGET_TOKEN_NOT_FOUND: &str = "Target token not found";
+
+pub struct RandomBytes; //TransferEthToNear
+
+impl RandomBytes {
+    /// Random bytes precompile address
+    ///
+    /// Address: `0xf861511815955326b953fa97b6955a2f8020a4e9`
+    /// This address is computed as: `&keccak("randomBytes")[12..]`
+    pub const ADDRESS: Address =
+        super::make_address(0xf8615118, 0x15955326b953fa97b6955a2f8020a4e9);
+}
+
+impl Precompile for RandomBytes {
+    fn required_gas(_input: &[u8]) -> Result<u64, ExitError> {
+        Ok(costs::RANDOM_BYTES_GAS)
+    }
+
+    #[cfg(feature = "contract")]
+    fn run(
+        input: &[u8],
+        target_gas: Option<u64>,
+        _context: &Context,
+        _is_static: bool,
+    ) -> EvmPrecompileResult {
+        let cost = Self::required_gas(input)?;
+        if let Some(target_gas) = target_gas {
+            if cost > target_gas {
+                return Err(ExitError::OutOfGas);
+            }
+        }
+
+        Ok(
+            PrecompileOutput::without_logs(cost, sdk::random_seed().to_fixed_bytes().to_vec())
```

Comment:
I think we will need some very clear documentation around this precompile. It is not a random bytes generator in the sense that repeated calls to this address in the same transaction will return the same value. This is a per-block entropy source which could then be used to create a random sequence.

Speaking of documentation, do we have any documentation of the Aurora custom precompiles? If not we should make an issue to write this.

Reply:
I updated the docstring for now, but I agree that it would be super helpful to have a precompile documentation.